src/Digdir.Domain.Dialogporten.Application/Features/V1/ServiceOwner/Dialogs/Commands/Create/CreateDialogCommand.cs

```diff
@@ -105,4 +108,20 @@ public async Task<CreateDialogResult> Handle(CreateDialogCommand request, Cancel
             domainError => domainError,
             concurrencyError => throw new UnreachableException("Should never get a concurrency error when creating a new dialog"));
     }
+
+    private static List<string> GetServiceResourceReferences(CreateDialogDto request)
+    {
+        var serviceResourceReferences = new List<string> { request.ServiceResource };
+
+        static bool IsExternalResource(string? resource)
+        {
+            return resource is not null && resource.StartsWith(Constants.ServiceResourcePrefix, StringComparison.OrdinalIgnoreCase);
+        }
+
+        serviceResourceReferences.AddRange(from action in request.ApiActions where IsExternalResource(action.AuthorizationAttribute) select action.AuthorizationAttribute!);
+        serviceResourceReferences.AddRange(from action in request.GuiActions where IsExternalResource(action.AuthorizationAttribute) select action.AuthorizationAttribute!);
+        serviceResourceReferences.AddRange(from element in request.Elements where IsExternalResource(element.AuthorizationAttribute) select element.AuthorizationAttribute!);
```

Comment:
```suggestion
        serviceResourceReferences.AddRange(request.ApiActions
            .Where(action => IsExternalResource(action.AuthorizationAttribute))
            .Select(action => action.AuthorizationAttribute!));
        serviceResourceReferences.AddRange(request.GuiActions
            .Where(action => IsExternalResource(action.AuthorizationAttribute))
            .Select(action => action.AuthorizationAttribute!));
        serviceResourceReferences.AddRange(request.Elements
            .Where(element => IsExternalResource(element.AuthorizationAttribute))
            .Select(element => element.AuthorizationAttribute!));
```


Using LINQ method chain to match rest of code base